src/main/scala/encry/local/miner/EncryMiningWorker.scala

```diff
@@ -1,41 +1,34 @@
 package encry.local.miner
 
-import akka.actor.Actor
-import encry.EncryApp._
+import akka.actor.{Actor, ActorRef}
 import encry.consensus.{CandidateBlock, ConsensusSchemeReaders}
-import encry.stats.StatsSender.MiningEnd
+import encry.local.miner.EncryMiner.MinedBlock
+import encry.local.miner.EncryMiningWorker.{DropChallenge, MineBlock, NextChallenge}
 import encry.utils.ScorexLogging
-import encry.view.EncryNodeViewHolder.ReceivableMessages.LocallyGeneratedModifier
-import encry.view.state.StateMode
 
-class EncryMiningWorker(initialCandidate: CandidateBlock, myNumber: Int, numberOfWorkers: Int)
-  extends Actor with ScorexLogging {
+class EncryMiningWorker(miner: ActorRef, myNumber: Int, numberOfWorkers: Int) extends Actor with ScorexLogging {
 
-  case class MineBlock(nonce: Long)
-
-  var candidate: CandidateBlock = initialCandidate
+  override def receive: Receive = {
 
-  override def preStart(): Unit = {
-    log.info("Starting new mining worker: " + self.path)
-    context.system.scheduler.scheduleOnce(settings.node.miningDelay) {
-      self ! MineBlock(Long.MaxValue / numberOfWorkers * myNumber)
-    }
-  }
+    case NextChallenge(candidate: CandidateBlock) => self ! MineBlock(candidate, Long.MaxValue / numberOfWorkers * myNumber)
 
-  override def receive: Receive = {
-    case newCandidate: CandidateBlock => candidate = newCandidate
-    case MineBlock(nonce) => ConsensusSchemeReaders.consensusScheme.verifyCandidate(candidate, nonce) match {
+    case MineBlock(candidate: CandidateBlock, nonce: Long) => ConsensusSchemeReaders.consensusScheme.verifyCandidate(candidate, nonce) match {
       case Some(block) =>
         log.info(s"New block is found: $block on worker $self.")
         //statsSender ! MiningEnd(block.header, myNumber)
-        nodeViewHolder ! LocallyGeneratedModifier(block.header)
```

Comment:
.fold(self ! MineBlock(candidate, nonce + 1)){ block =>
          log.info(s"New block is found: $block on worker $self.")
          miner ! MinedBlock(block)
        }

Reply:
@garmonbozzzia done